eo-parser/src/main/java/org/eolang/parser/EoParserErrors.java

```diff
@@ -73,66 +67,76 @@ public void syntaxError(
-            } else if (names[EoParser.RULE_objects].equals(rule)) {
-                detailed =
-                    "We expected a list of objects here but encountered something unexpected";
-            } else {
-                detailed = msg;
-            }
-            msgs.add(new MsgLocated(line, position, detailed).formatted());
-            msgs.add(
-                new MsgUnderlined(
-                    this.lines.line(line),
-                    position,
-                    Math.max(token.getStopIndex() - token.getStartIndex(), 1)
-                ).formatted()
-            );
+            result = getDetailedInputMismatch((Parser) recognizer, msg);
+            underlined = this.getMsgUnderlined(symbol, line, position);
         } else {
-            msgs.add(new MsgLocated(line, position, msg).formatted());
-            msgs.add(this.lines.line(line));
+            result = msg;
+            underlined = this.lines.line(line);
         }
+        final List<String> msgs = List.of(
+            new MsgLocated(line, position, result).formatted(),
+            underlined
+        );
         this.errors.add(new ParsingException(error, line, msgs));
     }
 
     @Override
     public Iterator<ParsingException> iterator() {
         return this.errors.iterator();
     }
+
+    private static String getDetailedNoViableAlt(final Parser parser) {
+        final String rule = parser.getRuleInvocationStack().get(0);
+        final String[] names = parser.getRuleNames();
+        final String detailed;
+        if (names[EoParser.RULE_objects].equals(rule)) {
+            detailed = "Invalid object list declaration";
+        } else if (names[EoParser.RULE_metas].equals(rule)) {
+            detailed = "Invalid meta declaration";
+        } else if (names[EoParser.RULE_program].equals(rule)) {
+            detailed = "Invalid program declaration";
+        } else if (names[EoParser.RULE_bound].equals(rule)) {
+            detailed = "Invalid bound object declaration";
+        } else if (names[EoParser.RULE_object].equals(rule)) {
+            detailed = "Invalid object declaration";
+        } else {
+            detailed = "no viable alternative at input";
+        }
+        return detailed;
+    }
+
+    private static String getDetailedInputMismatch(final Parser parser, final String msg) {
+        final String rule = parser.getRuleInvocationStack().get(0);
+        final String[] names = parser.getRuleNames();
+        final String detailed;
+        if (names[EoParser.RULE_program].equals(rule)) {
+            detailed =
+                "We expected the program to end here but encountered something unexpected";
+        } else if (names[EoParser.RULE_objects].equals(rule)) {
+            detailed =
+                "We expected a list of objects here but encountered something unexpected";
+        } else {
+            detailed = msg;
+        }
+        return detailed;
+    }
+
+    private String getMsgUnderlined(
```

Comment:
@vl4ds4m It's better to use a single-word method names (e.g., `underlined()`)

https://www.yegor256.com/2018/08/22/builders-and-manipulators.html